blocks/init/src/Blocks/custom/featured-content/featured-content.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * Template for the Featured Content view.
+ *
+ * @package EightshiftBoilerplate
+ */
+
+use EightshiftBoilerplateVendor\EightshiftLibs\Helpers\Components;
+
+$manifest = Components::getManifest(__DIR__);
+
+$blockClass = $attributes['blockClass'] ?? '';
+$blockName = $attributes['blockName'] ?? '';
+
+$unique = Components::getUnique();
+
+$featuredContentPostType = Components::checkAttr('featuredContentPostType', $attributes, $manifest);
+$featuredContentTaxonomy = Components::checkAttr('featuredContentTaxonomy', $attributes, $manifest);
+$featuredContentTerms = Components::checkAttr('featuredContentTerms', $attributes, $manifest);
+$featuredContentPosts = Components::checkAttr('featuredContentPosts', $attributes, $manifest);
+$featuredContentExcludeCurrentPost = Components::checkAttr('featuredContentExcludeCurrentPost', $attributes, $manifest);
+$featuredContentUseCurrentTerm = Components::checkAttr('featuredContentUseCurrentTerm', $attributes, $manifest);
+$featuredContentServerSideRender = Components::checkAttr('featuredContentServerSideRender', $attributes, $manifest);
+$featuredContentRandomOrder = Components::checkAttr('featuredContentRandomOrder', $attributes, $manifest);
+$featuredContentLayoutTotalItems = Components::checkAttr('featuredContentLayoutTotalItems', $attributes, $manifest);
+$featuredContentLoadMoreUse = Components::checkAttr('featuredContentLoadMoreUse', $attributes, $manifest);
+
+global $post;
+
+$args = [
+	'post_type' => $featuredContentPostType,
+	'posts_per_page' => $featuredContentLayoutTotalItems,
+	'fields' => 'ids',
+];
+
+if ($featuredContentTaxonomy) {
+	$args['tax_query'][0] = [
+		'taxonomy' => $featuredContentTaxonomy,
+		'field' => 'id',
+	];
+
+	if ($featuredContentTerms) {
+		$args['tax_query'][0]['terms'] = array_map(
+			function ($item) {
+				return $item['value'];
+			},
+			(array)$featuredContentTerms
+		);
+	} elseif ($featuredContentUseCurrentTerm && $post instanceof WP_Post && !$featuredContentServerSideRender) {
+		$currentTerms = get_the_terms($post->ID, $featuredContentTaxonomy);
+
+		if ($currentTerms) {
+			$args['tax_query'][0]['terms'] = [$currentTerms[0]->term_id]; // @phpstan-ignore-line
+		}
+	} else {
+		$args['tax_query'][0]['operator'] = 'NOT IN'; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+	}
+}
+
+$excludeList = [];
+
+if ($featuredContentExcludeCurrentPost && $post instanceof WP_Post) {
+	$excludeList[] = $post->ID;
+}
+
```

Comment:
It would be much better if we could have this inside a class method. Ideally this will be a filter that takes multiple arguments (as WP_Query args) and then just returns the query object in `$mainQuery` variable.

Reply:
💯

Reply:
agreed but this will overcomplicate the implementation so I would like to not do that.

Reply:
i have opened an issue https://github.com/infinum/eightshift-frontend-libs/issues/631